aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -74,103 +74,145 @@ void filterEngineConfigs(
   std::array<cudnn_frontend::Operation const *, 1> ops = {&op};
 
   auto opGraph = cudnn_frontend::OperationGraphBuilder()
       .setHandle(handle)
       .setOperationGraph(1, ops.data())
       .build();
-  // std::cout << opGraph.describe() << std::endl;
-
   auto heuristics = cudnn_frontend::EngineHeuristicsBuilder()
       .setOperationGraph(opGraph)
       .setHeurMode(CUDNN_HEUR_MODE_INSTANT)
       .build();
   auto fallback = cudnn_frontend::EngineFallbackListBuilder()
                     .setOperationGraph(opGraph)
-                    .setOperation(CUDNN_BACKEND_OPERATION_CONVOLUTION_FORWARD_DESCRIPTOR)
+                    .setOperation(desc)
                     .build();
-
   auto& engine_configs = heuristics.getEngineConfig(heuristics.getEngineConfigCount());
   auto& fallback_list = fallback.getFallbackList();
 
-  cudnn_frontend::EngineConfigList filtered_configs;
-  filterEngineConfigs(engine_configs, filtered_configs, deterministic, allow_tf32, input.scalar_type());
-  filterEngineConfigs(fallback_list, filtered_configs, deterministic, allow_tf32, input.scalar_type());
+  filterEngineConfigs(engine_configs, filtered_configs, deterministic, allow_tf32, x.scalar_type());
+  filterEngineConfigs(fallback_list, filtered_configs, deterministic, allow_tf32, x.scalar_type());
+}
 
+void try_filtered_configs(const cudnn_frontend::EngineConfigList filtered_configs, const CacheKey key, const cudnnHandle_t handle, const Tensor& x, const Tensor& y, const Tensor& w) {
   for (auto &cfg : filtered_configs) {
     try {
-      run(cfg);
+      run_conv_cfg(handle, x, y, w, cfg);
       engine_cache[key] = cfg;
       return;
     } catch (cudnn_frontend::cudnnException &e) {} catch(CuDNNError &e) {}
   }
   TORCH_CHECK(false, "Unable to find an engine to execute this computation");
 }
 
+void run_single_conv(const cudnnBackendDescriptorType_t operation,
+  const Tensor& x, const Tensor& y, const Tensor& w,
+  IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
+  bool benchmark, bool deterministic, bool allow_tf32) {
+  TORCH_CHECK(!benchmark, "not supported yet");
+
+  cudnnHandle_t handle = getCudnnHandle();
+
+  CacheKey key;
+  get_cachekey(key, operation, y, x, w, padding, stride, dilation, groups, deterministic, allow_tf32);
+  auto search = engine_cache.find(key);
+  if (search != engine_cache.end()) {
+    run_conv_cfg(handle, x, y, w, search->second);
+    return;
+  }
+
+  cudnn_frontend::EngineConfigList filtered_configs;
+  get_configs_from_heuristics(filtered_configs, handle, operation, x, y, w, key, padding, stride, dilation, deterministic, allow_tf32);
+
+  try_filtered_configs(filtered_configs, key, handle, x, y, w);
+}
+
+void raw_cudnn_convolution_forward_out(
+    const Tensor& output, const Tensor& input, const Tensor& weight,
+    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
+    bool benchmark, bool deterministic, bool allow_tf32)
+{
+  if (y.numel() > 0) {
```

Comment:
Did you test your change? `y` is not defined here.

Reply:
wow, can't believe I missed that, thanks
I think I restarted my environment and forgot to reset the experimental flag so I was rebuilding `Conv_v7.cpp` over and over...